lib/app/features/feed/views/components/text_editor/components/custom_blocks/text_editor_separator_block/text_editor_separator.dart

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/feed/views/components/text_editor/components/custom_blocks/text_editor_separator_block/text_editor_separator_close.dart';
+
+class TextEditorSeparator extends HookWidget {
+  const TextEditorSeparator({
+    required this.onRemove,
+    super.key,
+  });
+
+  final VoidCallback onRemove;
+
+  @override
+  Widget build(BuildContext context) {
+    final isSelected = useState(false);
+
+    final separatorWidth = 58.0.s;
+    final separatorHeight = 1.0.s;
+    final smallDotSize = 3.0.s;
+    final bigDotSize = 4.0.s;
+    final dotsPadding = 6.0.s;
+    final separatorsPadding = 10.0.s;
+
+    return GestureDetector(
+      onTap: () {
+        isSelected.value = !isSelected.value;
+      },
+      child: Center(
+        child: Padding(
+          padding: EdgeInsets.symmetric(vertical: 16.0.s),
+          child: Container(
+            width: 217.0.s,
+            height: 24.0.s,
+            decoration: BoxDecoration(
+              color: isSelected.value
+                  ? context.theme.appColors.primaryAccent.withOpacity(0.15)
+                  : Colors.transparent,
+              borderRadius: BorderRadius.circular(16.0.s),
+            ),
+            child: Stack(
+              alignment: Alignment.center,
+              children: [
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    Container(
+                      width: separatorWidth,
+                      height: separatorHeight,
+                      color: context.theme.appColors.onTertararyBackground,
+                    ),
+                    SizedBox(width: separatorsPadding),
+                    _buildDot(context, smallDotSize),
+                    SizedBox(width: dotsPadding),
+                    _buildDot(context, bigDotSize),
+                    SizedBox(width: dotsPadding),
+                    _buildDot(context, smallDotSize),
+                    SizedBox(width: separatorsPadding),
+                    Container(
+                      width: separatorWidth,
+                      height: separatorHeight,
+                      color: context.theme.appColors.onTertararyBackground,
+                    ),
+                  ],
+                ),
+                if (isSelected.value)
+                  Align(
+                    alignment: Alignment.topRight,
+                    child: TextEditorSeparatorClose(
+                      onPressed: onRemove,
+                    ),
+                  ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildDot(BuildContext context, double size) {
```

Comment:
Create please a private class and replace `Container` with `SizedBox.square(size)`

Reply:
```
SizedBox.square(size) {
 child: DecoratedBox(...),
}
```